File: Scala/src/main/scala/FizzBuzz.scala
object FizzBuzz {

  def eval(n: Int) = {
    (n % 3, n % 5) match {
      case (0, 0) => "FizzBuzz"
      case (0, _) => "Fizz"
      case (_, 0) => "Buzz"
      case _ => n.toString
    }
  }
}

File: Kotlin/src/main/kotlin/FizzBuzz.kt
object FizzBuzz {
    fun eval(n: Int): String {
        val (a, b) = Pair(n % 3, n % 5)
        if (a == 0 && b == 0) return "FizzBuzz"
        if (a == 0) return "Fizz"
        if (b == 0) return "Buzz"
        // こんな感じの値でのマッチは書けない？
        //when (Pair(n % 3, n % 5)) {
        //    Pair(0, 0) -> return "FizzBuzz"
        //    is #(0, *) -> return "Fizz"
        //    is #(*, *) -> return "Fizz"
        //}
        return n.toString()
    }
}
